### PR TITLE
Add error explanation for E0755

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @pickfire

### CODE REVIEW

File: compiler/rustc_error_codes/src/error_codes/E0755.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+effects except for their return value. You can find more information about it in
+the [unstable Rust Book].
+
+```
+#![feature(ffi_pure)]
+
+extern "C" {
+    #[ffi_pure] // ok!
+    pub fn strlen(s: *const i8) -> isize;
+}
```
Comment (pickfire) [inline compiler/rustc_error_codes/src/error_codes/E0755.md:23]: I believe this also work right?
```rust
#[ffi_pure]
extern "C" pub fn foo() {}
```

File: compiler/rustc_error_codes/src/error_codes/E0755.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#![feature(ffi_pure)]
+
+#[ffi_pure] // error!
+pub fn foo() {}
+# fn main() {}
+```
+
+The `ffi_pure` attribute can only be used on foreign functions which do not have
+effects except for their return value. You can find more information about it in
+the [unstable Rust Book].
```
Comment (pickfire) [inline compiler/rustc_error_codes/src/error_codes/E0755.md:15]:
```suggestion
side effects. You can find more information about it in the [unstable Rust Book].
```
Returning value is probably not counted as side effects.

File: compiler/rustc_error_codes/src/error_codes/E0755.md (hunk 1 above)
Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/error_codes/E0755.md:23]: I think it's better to have a bigger contrast. :)

File: compiler/rustc_error_codes/src/error_codes/E0755.md (hunk 2 above)
Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/error_codes/E0755.md]: I quoted the book.

**Comment by GuillaumeGomez** [comment]
Updated!

File: compiler/rustc_error_codes/src/error_codes/E0755.md (hunk 1 above)
Comment (pickfire) [reply compiler/rustc_error_codes/src/error_codes/E0755.md:22]: Oh, I was just suggesting to show both methods so the readers know.

File: compiler/rustc_error_codes/src/error_codes/E0755.md (hunk 2 above)
Comment (pickfire) [reply compiler/rustc_error_codes/src/error_codes/E0755.md]: Yes, I took from the book too.

Comment (GuillaumeGomez) [reply compiler/rustc_error_codes/src/error_codes/E0755.md]: Then perfect. :)

**Pull request merged**